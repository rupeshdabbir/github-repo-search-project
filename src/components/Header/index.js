// @flow

import React from 'react';
import {
  StyleSheet,
  css,
} from 'aphrodite/no-important';
import SearchForm from 'components/SearchForm';
import Logo from 'components/Logo';
import {viewport} from 'theme';
import 'suitcss-utils-flex/lib/flex-sm.css';

type Props = {
  onSubmit: Function,
  searchTerm: string,
};

export default function Header({onSubmit, searchTerm}: Props) {
  return (
    <div
      className={`${css(styles.Header)} u-sm-flex u-sm-flexJustifyBetween u-sm-flexAlignItemsCenter`}
    >
      <div className="u-sm-flexGrow2">
        <Logo />
      </div>
      <div className={`${css(styles.Header_form)} u-sm-flexGrow1`}>
        <SearchForm
          onSubmit={onSubmit}
          initialInputValue={searchTerm}
        />
      </div>
    </div>
  );
}

const styles = StyleSheet.create({
  Header: {
    paddingTop: 15,
    paddingBottom: 15,
  },

  Header_form: {
    marginTop: 20,

    [viewport.SM]: {
      margin: 0,
    },
  },
});
