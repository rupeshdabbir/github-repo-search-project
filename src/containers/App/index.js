// @flow

import React from 'react';
import { HashRouter as Router, Redirect, Switch } from 'react-router-dom';
import SearchScreen from 'screens/Search';
import DefaultLayout from 'screens/Default';

import 'css/global.css';

export default function App() {
  return (
    <Router>
      <Switch>
        <DefaultLayout exact={true} path="/search" component={SearchScreen} />
        <Redirect to="/search" />
      </Switch>
    </Router>
  );
}
