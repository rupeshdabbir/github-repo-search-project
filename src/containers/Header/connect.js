// @flow

import { connect } from 'react-redux';

export function mapDispatchToProps(dispatch: Function) {
  return {
    searchForRepos(search: string) {
      dispatch({
        type: 'SEARCH_REQUEST',
        payload: {
          search,
        },
      });
    },
  };
}

export default connect(null, mapDispatchToProps);
