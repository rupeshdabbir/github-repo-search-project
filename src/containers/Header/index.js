// @flow

import React, { Component } from 'react';
import { StyleSheet } from 'aphrodite/no-important';
import Header from 'components/Header';
import Container from 'components/Container';
import connect from './connect';

type Props = {
  onSubmit: Function,
  searchQuery: string,
  searchTerm: string,
  searchForRepos: Function,
};

export class HeaderContainer extends Component {
  static defaultProps = {
    searchTerm: '',
  };

  constructor(props: Props) {
    super(props);
    this.handleSearchRepos(props.searchQuery);
  }

  componentWillReceiveProps(nextProps: Props) {
    if (nextProps.searchQuery !== this.props.searchQuery) {
      this.handleSearchRepos(nextProps.searchQuery);
    }
  }

  handleSearchRepos(searchQuery: string): void {
    if (!searchQuery) {
      return;
    }
    this.props.searchForRepos(searchQuery);
  }

  render() {
    const { onSubmit, searchTerm } = this.props;

    return (
      <Container rootStyle={styles.HeaderContainer}>
        <Header onSubmit={onSubmit} searchTerm={searchTerm} />
      </Container>
    );
  }
}

const styles = StyleSheet.create({
  HeaderContainer: {
    backgroundColor: '#f4f4f4',
    borderBottom: '1px solid #aaa',
    boxShadow: '0px 0px 2px 1px rgba(0, 0, 0, .2)',
  },
});

export default connect(HeaderContainer);
