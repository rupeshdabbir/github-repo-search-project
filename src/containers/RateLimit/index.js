import React, { Component } from 'react';
import { css, StyleSheet } from 'aphrodite/no-important';
import 'suitcss-utils-flex/lib/flex.css';
import RateLimit from 'components/RateLimit';
import isUndefined from 'lodash/fp/isUndefined';
import connect from './connect';

type Props = {
  coreLimit: ?number,
  dispatch: Function,
  searchLimit: ?number,
  searchRemaining: ?number,
  searchReset: ?string,
};

export class RateLimitContainer extends Component {
  constructor(props: Props) {
    super(props);
    const { coreLimit, searchLimit, dispatch } = props;

    if (isUndefined(coreLimit) || isUndefined(searchLimit)) {
      dispatch({ type: 'API_RATE_LIMIT_REQUEST' });
    }
  }

  render() {
    const { searchLimit, searchRemaining, searchReset } = this.props;

    return (
      <div className="u-flex u-flexAlignItemsCenter">
        <p className={css(styles.RateLimitContainer_label)}>API requests</p>
        <div className={`${css(styles.RateLimitContainer)} u-flex`}>
          <div className={css(styles.RateLimitContainer_item)}>
            <RateLimit label="Search" remaining={searchRemaining} limit={searchLimit} reset={searchReset} />
          </div>
        </div>
      </div>
    );
  }
}

const styles = StyleSheet.create({
  RateLimitContainer: {
    backgroundColor: '#333',
    color: '#eee',
    padding: '5px 8px',
    borderRadius: 5,
  },

  RateLimitContainer_label: {
    fontSize: 14,
    marginRight: 10,
  },

  RateLimitContainer_item: {
    ':first-child': {
      marginRight: 5,
    },

    ':last-child': {
      marginLeft: 5,
    },
  },
});

export default connect(RateLimitContainer);
