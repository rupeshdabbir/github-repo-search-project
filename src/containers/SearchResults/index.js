// @flow

import React from 'react';
import { css, StyleSheet } from 'aphrodite/no-important';
import SearchTable from 'components/SearchTable';
import Pagination from 'components/Pagination';
import ResultsMessage from 'components/ResultsMessage';
import ErrorBoundary from 'components/ErrorBoundary';
import Loading from 'components/Loading';
import Container from 'components/Container';
import connect from './connect';

type Props = {
  isPending: boolean,
  pagination: ?Object,
  searchTerm: string,
  totalResults: number,
  userIds: Array<number>,
};

export function SearchResultsContainer(props: Props) {
  const { isPending, pagination, searchTerm, totalResults, userIds, totalResponse } = props;

  console.log('props', props);

  if (!searchTerm) {
    return null;
  }

  if (isPending) {
    return (
      <div className={css(styles.SearchResultsContainer_wrapLoading)}>
        <Loading />
      </div>
    );
  }

  return (
    <Container>
      <div className={css(styles.SearchResultsContainer_item, styles.SearchResultsContainer_wrapResultsMessage)}>
        <ErrorBoundary>
          <ResultsMessage searchTerm={searchTerm} resultsTotal={totalResults} pageTotal={userIds.length} />
        </ErrorBoundary>
      </div>
      <div className={css(styles.SearchResultsContainer_item)}>
        <ErrorBoundary>
          {/* <SearchResults searchTerm={searchTerm} ids={userIds} entities={userEntities} total={totalResults} /> */}
          <SearchTable totalResponse={totalResponse} />
        </ErrorBoundary>
      </div>
      <div className={css(styles.SearchResultsContainer_item)}>
        <ErrorBoundary>
          <Pagination {...pagination} />
        </ErrorBoundary>
      </div>
    </Container>
  );
}

SearchResultsContainer.defaultProps = {
  pagination: {},
  searchTerm: '',
};

const styles = StyleSheet.create({
  SearchResultsContainer_wrapResultsMessage: {
    marginBottom: 30,
  },

  SearchResultsContainer_wrapLoading: {
    marginTop: 25,
  },

  SearchResultsContainer_item: {
    marginTop: 15,
  },
});

export default connect(SearchResultsContainer);
