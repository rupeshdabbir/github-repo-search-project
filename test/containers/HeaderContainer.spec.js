import React from 'react';
import { shallow } from 'enzyme';
import { StyleSheetTestUtils } from 'aphrodite';

import { HeaderContainer } from 'containers/Header';

describe('Component: HeaderContainer', () => {
  beforeAll(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });

  afterAll(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  describe('when the searchQuery prop is not present', () => {
    it('should not call the search action', () => {
      const spy = jest.fn();
      shallow(<HeaderContainer searchQuery={''} searchForRepos={spy} onSubmit={jest.fn()} />);
      expect(spy).not.toHaveBeenCalled();
    });
  });

  describe('when the searchQuery prop is present', () => {
    it('should call the search action', () => {
      const spy = jest.fn();
      shallow(<HeaderContainer searchQuery={'?q=test'} searchForRepos={spy} onSubmit={jest.fn()} />);
      expect(spy.mock.calls[0]).toMatchSnapshot();
    });
  });

  describe('when the searchQuery prop changes', () => {
    it('should call the search action with the new value', () => {
      const spy = jest.fn();
      const wrapper = shallow(<HeaderContainer searchQuery={'?q=test'} searchForRepos={spy} onSubmit={jest.fn()} />);
      expect(spy.mock.calls[0]).toMatchSnapshot();
      wrapper.setProps({ searchQuery: '?q=foobar' });
      expect(spy.mock.calls[1]).toMatchSnapshot();
    });
  });
});
