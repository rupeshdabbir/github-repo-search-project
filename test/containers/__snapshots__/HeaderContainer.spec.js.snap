// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: HeaderContainer when the searchQuery prop changes should call the search action with the new value 1`] = `
Array [
  "?q=test",
]
`;

exports[`Component: HeaderContainer when the searchQuery prop changes should call the search action with the new value 2`] = `
Array [
  "?q=foobar",
]
`;

exports[`Component: HeaderContainer when the searchQuery prop is present should call the search action 1`] = `
Array [
  "?q=test",
]
`;
